Annotated variant

victoria34 opened the pull request.

victoria34 committed:
Implement oncokb_match() in pre-run way

victoria34 committed:
Remove upsert fields UNIQUE_CLINICAL_ID and UNIQUE_GENOMIC_ID

victoria34 committed:
1. Add "--drop" in args for dropping existing clinical and genomic collection.
2. Create dump_collection() and restore_collection() util functions

victoria34 committed:
Update code after rebasing from dfci master branch

victoria34 committed:
Remove useless code

victoria34 committed:
Remove temp solution for main_type

victoria34 committed:
Remove no_hugo_symbol

zhx828 commented on matchengine.py:
```diff
@@ -299,6 +329,7 @@ def match(args):
     param_outpath_help = 'Destination and name of your results file.'
     param_trial_format_help = 'File format of input trial data. Default is YML.'
     param_patient_format_help = 'File format of input patient data (both clinical and genomic files). Default is CSV.'
+    param_match_method_help = 'Match method name. The default is oncokb and uses oncokb_match().'
```
The default is `OncoKB`

zhx828 commented on matchengine/engine.py:
```diff
@@ -220,103 +229,227 @@ def create_trial_tree(self, raw_data, no_validate=False):
+            # remove "!"
+            hugo_symbol = hugo_symbol[1:]
+
+        if annotated_variant.startswith('!'):
```
not logic makes the whole matching to false according to the following statement. What happens when hugo: `BRAF`, annotated_variant: `!V600E`?

victoria34 replied:
I think we discussed it before. If one of them has `Not`, the matching of this match clause will be negative query. For `BRAF !V600E`, the match result will be (all_trials - trials matched for BRAF V600E).

victoria34 committed:
Fix bugs based Kelsey's data

victoria34 committed:
Check if hugoSymbol and proteinChange are in genomic data

victoria34 committed:
Fix "All solid tumors" issue

victoria34 committed:
Add "All Liquid Tumors"

victoria34 committed:
Remove dup alteration from final_genomic_infos

victoria34 committed:
Run general_match() when genomic['annotated_variant'] = 'wildtype'

victoria34 committed:
Check genomic_id exists

ethansiegl committed:
temporary fixes for memory issues (#19)

* When inserting trial matches into the db, do 1000 objects at a time instead of as one big blob
* Move conversion of the trial_match object to dataframe up a level. Do this so garbage collection works and a large object is not left hanging around in memory

victoria34 committed:
Display protocol_no or nct_id

ethangk committed:
Updated mongoexport to use the mongo uri (#15)

please ignore earlier comment on issue did not see this PR. thanks for the update!

ethansiegl committed:
Revert "Updated mongoexport to use the mongo uri (#15)" (#22)

This reverts commit ce7914156af49d5d1271f2d380b7b1748c6b7521.

ethansiegl committed:
re merge previous reverted commit (accident) (#23)

ethansiegl committed:
add timestamp logging (#24)

victoria34 committed:
Add clinical matching info to trial_match

“trial_oncotree_primary_diagnosis”, “trial_age_numerical”

victoria34 committed:
Fix oncokb_api_match()

victoria34 committed:
Remove Not logic itself from db.genomic

victoria34 committed:
Import clinical info to matched result(inner join)

Make inner join between final_clinical_infos and final_genomic_infos

victoria34 committed:
Remove useless function

victoria34 committed:
Optimize code

victoria34 committed:
Optimize code

victoria34 committed:
Add clinical_info to genomic_info

victoria34 committed:
Support main type query

victoria34 committed:
Remove useless code
